cpp/util/libevent_wrapper.cc

```diff
@@ -383,6 +386,79 @@ void HttpServer::HandleRequest(evhttp_request* req, void* userdata) {
 }
 
 
+QueryParams ParseQuery(evhttp_request* req) {
+  evkeyvalq keyval;
+  QueryParams retval;
+
+  // We return an empty result in case of a parsing error.
+  if (evhttp_parse_query_str(evhttp_uri_get_query(
+                                 evhttp_request_get_evhttp_uri(req)),
+                             &keyval) == 0) {
+    for (evkeyval* i = keyval.tqh_first; i; i = i->next.tqe_next) {
+      retval.insert(make_pair(i->key, i->value));
+    }
+  }
+
+  return retval;
+}
+
+
+bool GetParam(const QueryParams& query, const string& param, string* value) {
+  CHECK_NOTNULL(value);
+
+  auto it = query.find(param);
```

Comment:
This here is the only actual change (it used to be `multimap<string, string>::const_iterator`), beside the `QueryParams` typedef. I lifted it straight out of #992, but it's now more equivalent to `QueryParams::iterator` than `QueryParams::const_iterator`. `const auto` wouldn't work, as it would prevent advancing the iterator.

I think it's all right like this, though, the rest of the function is short enough to be able to comprehensively review it.